src/requestSchemas.ts

```diff
@@ -0,0 +1,201 @@
+import { constants } from '@projecttacoma/node-fhir-server-core';
+import { z } from 'zod';
+import { BadRequestError, NotImplementedError } from './util/errorUtils';
+
+const DATE_REGEX = /([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1]))?)/;
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#package
+const UNSUPPORTED_PACKAGE_ARGS = [
+  'capability',
+  'check-system-version',
+  'count',
+  'force-system-version',
+  'include-components',
+  'include-dependencies',
+  'manifest',
+  'offset',
+  'system-version'
+];
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#requirements
+const UNSUPPORTED_DATA_REQ_ARGS = [
+  'expression',
+  'include-dependencies',
+  'include-components',
+  'manifest',
+  'parameters'
+];
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#search
+const UNSUPPORTED_CORE_SEARCH_ARGS = [
+  'composed-of',
+  'context',
+  'context-type',
+  'context-type-quantity',
+  'context-type-value',
+  'date',
+  'depends-on',
+  'derived-from',
+  'effective',
+  'jurisdiction',
+  'predecessor',
+  'successor',
+  'topic'
+];
+
+const UNSUPPORTED_LIBRARY_SEARCH_ARGS = [...UNSUPPORTED_CORE_SEARCH_ARGS, 'content-type', 'type'];
+
+const UNSUPPORTED_MEASURE_SEARCH_ARGS = [...UNSUPPORTED_CORE_SEARCH_ARGS, 'publisher'];
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
```

Comment:
Discussed on a call, but this error map breaks things. @natjoe4 is looking into solutions